src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/CommandSettingsViewModel.cs

```diff
@@ -2,18 +2,25 @@
 // The Microsoft Corporation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using ManagedCommon;
 using Microsoft.CmdPal.UI.ViewModels.Models;
 using Microsoft.CommandPalette.Extensions;
 
 namespace Microsoft.CmdPal.UI.ViewModels;
 
-public partial class CommandSettingsViewModel(ICommandSettings _unsafeSettings, CommandProviderWrapper provider, TaskScheduler mainThread)
+public partial class CommandSettingsViewModel(ICommandSettings? _unsafeSettings, CommandProviderWrapper provider, TaskScheduler mainThread)
 {
     private readonly ExtensionObject<ICommandSettings> _model = new(_unsafeSettings);
 
     public ContentPageViewModel? SettingsPage { get; private set; }
 
-    public void InitializeProperties()
+    public bool Initialized { get; private set; }
+
+    public bool HasSettings =>
+        _model.Unsafe != null && // We have a settings model AND
```

Comment:
I don't love this call to _model.Unsafe, it seems icky to do it in a public method where they don't know there's an unsafe call. Can the call inside `UnsafeInitializeProperties` set a public property that means the same thing on this class? Or otherwise make this safe?

Reply:
ah okay - _this_ by itself is safe, but I'm glad it caught your eye (that's what it was designed to do!)

`_model` is an `ExtensionObject<T>`, which has one member - `T? Unsafe`. Just checking if `.Unsafe != null` is by itself, safe. It's just that doing anything more with `.Unsafe` (i.e. `foo.Unsafe?.Bar()`) is _not_ safe (because the `T` is some object that might be out-of-proc and we want to make sure we deal with that).

and I named it `Unsafe` because I thought that any reviewer would see that and immediately doubletake on it and make sure that it was being used safely. 

:mission_accomplished_banner:

Reply:
This made me lol. Sounds good. We may or may not in the future want to implement some of the hullabaloo Amir was putting into Dev Home around ensuring we're using alive out-of-proc objects and whatnot, probably cleaner in its own way, but definitely more complex. I had a medium-complicated version of it for widget stuff.